### ISSUE TITLE
Inventory is not correctly populated after provisioning with libvirt

### ISSUE THREAD

**Issue by dyasny** [CentOS-PaaS-SIG/linchpin#540]
Here's what the IPs are:
```
# for i in `virsh list --name`; do echo $i; virsh domifaddr $i; done
master_0
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet0      52:54:00:a0:5b:d8    ipv4         192.168.122.12/24
 
node_0
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet1      52:54:00:e4:01:d7    ipv4         192.168.122.250/24
 
node_1
 Name       MAC address          Protocol     Address
-------------------------------------------------------------------------------
 vnet2      52:54:00:97:79:d4    ipv4         192.168.122.10/24
```
And here's the resulting inventory:
```
[nodes]
192.168.122.10 hostname=192.168.122.10
192.168.122.250 hostname=192.168.122.250
192.168.122.12 hostname=192.168.122.12
 
[nfs]
192.168.122.10 hostname=192.168.122.10
 
[masters]
192.168.122.10 hostname=192.168.122.10
```

As you can see above, master is 192.168.122.12 ad the assigned master is in fact a node - 192.168.122.10

**Comment by samvarankashyap** [comment]
@dyasny could you please share the topology file and layout file you have used to create the above inventory ?

**Comment by dyasny** [comment]
```yaml
inventory_layout:
#  inventory_file: cnv-inventory.inv
  vars:
    hostname: __IP__
  hosts:
    "master":
      count: 1
      host_groups:
        - masters
        - etcd
        - nodes
        - nfs
    "node":
      count: 2
      host_groups:
        - nodes
  host_groups:
    OSEv3:
      children:
        - masters
        - nodes
        - etcd
        - nfs
      vars:
        ansible_ssh_user: root
        openshift_schedulable: true
        openshift_node_labels: "{'region': 'infra','zone': 'default', 'node-role.kubernetes.io/compute': 'true'}"
        openshift_deployment_type: openshift-enterprise
        openshift_clock_enabled: true
        openshift_master_identity_providers: "[{'name': 'allow_all_auth', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]"
        openshift_disable_check: "memory_availability,disk_availability"
        openshift_enable_excluders: false
        template_service_broker_install: false
        openshift_use_manageiq: false
        openshift_install_examples: false
 
        openshift_hosted_etcd_storage_kind: nfs
        openshift_hosted_etcd_storage_nfs_options: "*(rw,root_squash,sync,no_wdelay)"
        openshift_hosted_etcd_storage_nfs_directory: "/opt/osev3-etcd"
        openshift_hosted_etcd_storage_volume_name: "etcd-vol2"
        openshift_hosted_etcd_storage_access_modes: "['ReadWriteOnce']"
        openshift_hosted_etcd_storage_volume_size: "1100M"
        openshift_hosted_etcd_storage_labels: "{'storage': 'etcd'}"
 
        ansible_service_broker_refresh_interval: "20s"
        ansible_service_broker_local_registry_whitelist: "[\".*-apb$\"]"
        ansible_service_broker_image_prefix: "registry.access.stage.redhat.com/openshift3/ose-"
        ansible_service_broker_image_tag: "latest"
        ansible_service_broker_registry_url: "registry.access.stage.redhat.com"
 
        ansible_service_broker_etcd_image_prefix: "quay.io/coreos/"
        ansible_service_broker_etcd_image_tag: "latest"
 
        docker_dev: "/dev/vdb"
 
        oreg_url: "registry.access.stage.redhat.com/openshift3/ose-${component}:${version}"
        openshift_docker_additional_registries: "registry.access.stage.redhat.com"
        openshift_docker_insecure_registries: "registry.access.stage.redhat.com"
        openshift_service_catalog_image_prefix: "registry.access.stage.redhat.com/openshift3/ose-"
        template_service_broker_prefix: "registry.access.stage.redhat.com/openshift3/ose-"
        openshift_web_console_prefix: "registry.access.stage.redhat.com/openshift3/ose-"
 
        osm_default_node_selector: ""
 
 
 
 
 
 
---
topology_name: cnv_libvirt_cluster
resource_groups:
  - resource_group_name: libvirt
    resource_group_type: libvirt
    resource_definitions:
      - role: libvirt_node
        name: master
        uri: qemu:///system
        count: 1
        image_src: file:///var/lib/libvirt/images/base_image
        memory: 8196
        vcpus: 2
        arch: x86_64
        ssh_key: id_rsa
        networks:
          - name: default
      - role: libvirt_node
        name: node
        uri: qemu:///system
        count: 2
        image_src: file:///var/lib/libvirt/images/base_image
        memory: 8196
        vcpus: 4
        arch: x86_64
        ssh_key: id_rsa
        networks:
          - name: default
```

**Comment by samvarankashyap** [comment]
@dyasny could you please test #581 PR to see if it fixes this issue or not .

**Comment by lukas-bednar** [comment]
@samvarankashyap It looks like that linchpin 1.6.0 introduced this issue again. I see this behav there.

**Comment by p3ck** [comment]
this still isn't right...

```yaml
---
topology_name: "bkr-new"
resource_groups:
  - resource_group_name: "bkr-new"
    resource_group_type: beaker
    resource_definitions:
      - role: bkr_server
        # option to set job whiteboard message
        whiteboard: Provisioned with linchpin Job 1
        # options to configure amount of time spent polling beaker:
        # 60 attempts with 60 seconds wait time between them,
        # provisioning timeout is roughly 3600 seconds
        max_attempts: 240
        attempt_wait_time: 60
        # option to set cancellation message if linchpin cancels provisioning
        cancel_message: Job cancelled on account of rain
        recipesets:
          - distro: RHEL-6.5
            name: bkr-job1
            arch: x86_64
            count: 1
  - resource_group_name: "bkr-second"
    resource_group_type: beaker
    resource_definitions:
      - role: bkr_server
        # option to set job whiteboard message
        whiteboard: Provisioned with linchpin Job 2
        # options to configure amount of time spent polling beaker:
        # 60 attempts with 60 seconds wait time between them,
        # provisioning timeout is roughly 3600 seconds
        max_attempts: 240
        attempt_wait_time: 60
        # option to set cancellation message if linchpin cancels provisioning
        cancel_message: Job cancelled on account of rain
        recipesets:
          - distro: RHEL-6.4
            name: bkr-job2
            arch: x86_64
            count: 1
```
The above produce the following inventory:
[all]
amd-pence-01.lab.bos.redhat.com hostname=amd-pence-01.lab.bos.redhat.com
hp-ml110g7-01.khw.lab.eng.bos.redhat.com hostname=hp-ml110g7-01.khw.lab.eng.bos.redhat.com

amd = rhel6.4
hp = rhel6.5

Yet the following topology does work correctly:
```yaml
topology_name: "test-rhel7"
resource_groups:
  - resource_group_name: "test-rhel7"
    resource_group_type: beaker
    resource_definitions:
      - role: bkr_server
        whiteboard: Provisioned with linchpin
        max_attempts: 240
        attempt_wait_time: 60
        cancel_message: Job cancelled on account of rain
        recipesets:
          - distro: RHEL-7.4
            arch: x86_64
            variant: Server
            count: 1
            name: rhel74-1
            keyvalue:
              - DISKSPACE>=100000
            bkr_data: lvm-on
            hostrequires:
              - tag: arch
                op: "="
                value: "X86_64"
              - tag: memory
                op: ">="
                value: "1000"
          - distro: RHEL-7.5
            arch: x86_64
            variant: Server
            count: 1
            name: rhel75-2
            keyvalue:
              - DISKSPACE>=100000
            bkr_data: lvm-off
            hostrequires:
              - tag: arch
                op: "="
                value: "X86_64"
              - tag: memory
                op: ">="
                value: "1000"
```

[all]
amd-pike-08.klab.eng.bos.redhat.com hostname=amd-pike-08.klab.eng.bos.redhat.com
cisco-c220m3-01.klab.eng.bos.redhat.com hostname=cisco-c220m3-01.klab.eng.bos.redhat.com

amd-pike = rhel-7.4
cisco-c220m3 = rhel-7.5

**Comment by lukas-bednar** [comment]
@samvarankashyap @herlo I think we have a regression, we see same issue again with linchpin==1.7.1.

**Comment by p3ck** [comment]
@lukas-bednar Please provide your pin and topology files so we can reproduce.  Also include the incorrect inventory file.

**Comment by lukas-bednar** [comment]
@p3ck here you go, I removed some internal references from layout, but I doubt it has any effect in reproducing this issue. [linchpin-files.txt](https://github.com/CentOS-PaaS-SIG/linchpin/files/2860668/linchpin-files.txt)

Here is generated [inventory.txt](https://github.com/CentOS-PaaS-SIG/linchpin/files/2860690/inventory.txt) , I added mapping fqdn to IPs at the end of inventory.

Here are resulting VMs reporting different roles as well [ocp-nodes.txt](https://github.com/CentOS-PaaS-SIG/linchpin/files/2860700/ocp-nodes.txt)

Note that it is not 100% reproducible, I could see it on 2 of 5 environments today.
Please let me know if you need any other info.

**Comment by 14rcole** [comment]
@lukas-bednar since I'm having some trouble reproducing it (and you said the issue is not 100% reproducible), can you give the the rundb files from both a successful and unsuccessful run? Or at least from an unsuccessful run?

**Comment by 14rcole** [comment]
@lukas-bednar I meant to say the linchpin.latest file, sorry

**Comment by 14rcole** [comment]
@lukas-bednar I'm still working on a fix but for now you can swap the two resource definitions in your topology and PinFile

**Comment by lukas-bednar** [comment]
@14rcole take your time ;-) ... I am glad that you were able to reproduce it !!!

**Comment by samvarankashyap** [comment]
fixed by #959
